tests/flows/test_vesting.py

```diff
@@ -147,32 +285,462 @@ def test_main(
+        # counter is used to count the number of times when get_vesting is called
+        # (it mints another block that change the formula)
+        counter += 1
+        vested_amount = side_staking.get_amount_vested_from_event(
+            tx_hash=tx_hash, erc20_token=erc20_token, from_wallet=publisher_wallet
+        )
+        assert (
+            vested_amount
+            == int(((block + 1 + counter) * vesting_amount) / vested_blocks)
+            - amount_vested_so_far
+        )
+        # claim tokens after vested blocks
+        assert dt_balance_before < erc20_token.balanceOf(
+            erc20_token.get_payment_collector()
+        )
+        assert (
+            erc20_token.balanceOf(erc20_token.get_payment_collector())
+            == vested_amount + amount_vested_so_far
+        )
+
+
+def test_vesting_publisher_exit_scam(
+    web3,
+    config,
+    publisher_wallet,
+    consumer_wallet,
+    another_consumer_wallet,
+    factory_router,
+    side_staking,
+):
+    """Tests publisher exit scam flow when the owner of the pool claimed the datatokens from vesting,
+    waits for others to add liquidity in his pool and then dumps the pool."""
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet, cap=to_wei(2000)
+    )
+
+    mint_fake_OCEAN(config)
+    vesting_amount = to_wei(18)
+    initial_ocean_liq = to_wei(200)
+
+    ocean_contract = ERC20Token(web3=web3, address=get_address_of_type(config, "Ocean"))
+    ocean_contract.approve(factory_router.address, initial_ocean_liq, publisher_wallet)
+
+    tx = erc20_token.deploy_pool(
+        rate=to_wei(1),
+        base_token_decimals=ocean_contract.decimals(),
+        vesting_amount=initial_ocean_liq // 100 * 9,
+        vesting_blocks=2500000,
+        base_token_amount=initial_ocean_liq,
+        lp_swap_fee_amount=to_wei("0.003"),
+        publish_market_swap_fee_amount=to_wei("0.001"),
+        ss_contract=get_address_of_type(config, "Staking"),
+        base_token_address=ocean_contract.address,
+        base_token_sender=publisher_wallet.address,
+        publisher_address=publisher_wallet.address,
+        publish_market_swap_fee_collector=get_address_of_type(
+            config, "OPFCommunityFeeCollector"
+        ),
+        pool_template_address=get_address_of_type(config, "poolTemplate"),
+        from_wallet=publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    pool_event = erc20_token.get_event_log(
+        ERC721FactoryContract.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    bpool = BPool(web3, pool_event[0].args.poolAddress)
+
+    # check if the vesting amount is correct
+    assert side_staking.get_vesting_amount(erc20_token.address) == vesting_amount
+
+    # check if vesting is correct
+    dt_balance_before = erc20_token.balanceOf(erc20_token.get_payment_collector())
```

Comment:
This method has several calls to `erc20_token.get_payment_collector()`, including inside larger expressions that black then forces across multiple lines, which in turn hurts readability. Recommend: create new line `pmt_collector = erc20_token.get_payment_collector()`, and use it throughout the function.